File: 1018-largest-perimeter-triangle/1018-largest-perimeter-triangle.cpp
class Solution {
public:
    int largestPerimeter(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int i = 0, j = 1, k = 2, ans = 0;
        while(k < nums.size()){
            if(nums[i] + nums[j] > nums[k]){
                ans = max(ans, nums[i] + nums[j] + nums[k]);
            }
            i++;
            j++;
            k++;
        }
        return ans;
    }
};

File: 1018-largest-perimeter-triangle/1018-largest-perimeter-triangle.py
class Solution:
    def largestPerimeter(self, A: List[int]) -> int:
        # sort side length in descending order
        A.sort( reverse = True )
        
		# Try and test from largest side length
        for i in range( len(A) - 2):
            
            if A[i] < A[i+1] + A[i+2]:
                # Early return when we find largest perimeter triangle
                return A[i] + A[i+1] + A[i+2]
        
        # Reject: impossible to make triangle
        return 0